modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -1236,136 +1237,93 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
           amount: currentAmount,
         });
-      });
-      explanation.outputAmount = spendAmount;
-      explanation.changeAmount = changeAmount;
-
-      // add fee info if available
-      if (params.feeInfo) {
-        explanation.displayOrder.push('fee');
-        explanation.fee = params.feeInfo;
-      }
-
-      if (_.isInteger(transaction.locktime) && transaction.locktime > 0) {
-        explanation.locktime = transaction.locktime;
-        explanation.displayOrder.push('locktime');
+        return;
       }
 
-      const unspentValues = {};
-
-      // get information on tx inputs
-      const inputSignatures = transaction.ins.map((input, idx) => {
-        const hasSigScript = !_.isEmpty(input.script);
-        const hasWitnessScript = !_.isEmpty(input.witness);
-
-        if (!hasSigScript && !hasWitnessScript) {
-          // no sig script or witness data for this input
-          debug('no signature script or witness script data for input %s', idx);
-          return 0;
-        }
-
-        let parsedSigScript;
-        try {
-          parsedSigScript = utxolib.bitgo.parseSignatureScript(transaction.ins[idx]);
-        } catch (e) {
-          return false;
-        }
+      spendAmount += currentAmount;
+      explanation.outputs.push({
+        address: currentAddress,
+        amount: currentAmount,
+      });
+    });
+    explanation.outputAmount = spendAmount;
+    explanation.changeAmount = changeAmount;
 
-        if (hasWitnessScript) {
-          if (!txInfo || !txInfo.unspents) {
-            // segwit txs require input values, cannot validate signatures
-            debug('unable to retrieve input amounts from unspents - cannot validate segwit input signatures');
-            return 0;
-          }
+    // add fee info if available
+    if (params.feeInfo) {
+      explanation.displayOrder.push('fee');
+      explanation.fee = params.feeInfo;
+    }
 
-          // lazily populate unspent values
-          if (_.isEmpty(unspentValues)) {
-            txInfo.unspents.forEach((unspent) => {
-              unspentValues[unspent.id] = unspent.value;
-            });
-          }
-        }
+    if (_.isInteger(transaction.locktime) && transaction.locktime > 0) {
+      explanation.locktime = transaction.locktime;
+      explanation.displayOrder.push('locktime');
+    }
 
-        const nonEmptySignatures = parsedSigScript.signatures?.filter((sig) => !_.isEmpty(sig)) ?? [];
-        const validSignatures = nonEmptySignatures.map((sig, sigIndex) => {
-          if (_.isEmpty(sig)) {
-            return false;
-          }
+    const prevOutputs = params.txInfo?.unspents.map((u) => toOutput(u, this.network));
 
-          const parentTxId = (Buffer.from(input.hash).reverse() as Buffer).toString('hex');
-          const inputId = `${parentTxId}:${input.index}`;
-          const amount = unspentValues[inputId];
+    // get information on tx inputs
+    const inputSignatures = transaction.ins.map((input, idx): number => {
```

Comment:
Input signatures represents the number of valid signatures for each input in the transaction right? I think that could be better explained in comments, either here or on the `TransactionExplanation` type. This had me confused for a while because I thought this was an array of the actual signatures, that's what the current name makes it seem like.

(Again I realize it's mostly old code, but might as well improve while we're here)

Reply:
will do in follow-up